src/iter.rs

```diff
@@ -24,10 +35,16 @@ impl<M: Miniconf + ?Sized, const L: usize, const TS: usize> Iterator for Minicon
     fn next(&mut self) -> Option<Self::Item> {
         let mut path = Self::Item::new();
 
-        if M::next_path(&mut self.state, &mut path) {
+        if M::next_path(&mut self.state, &mut path).unwrap() {
```

Comment:
Having this unwrap sucks. It's fine for our usecase, but introduces format bloat into applications that otherwise may not have it.

Some options:
1. Return a `Result<String, IterErr>` as the `Item`
2. match on the error and generate internal panic (doesn't seem much different than just unwrap IMO)
3. Somehow handle and log the error instead of failing
4. Leave it as is and indicate that this is a design limitation

I think (1) would be my preference? I'm not entirely sure. What are your thoughts?

Reply:
This PR replaces all the `unreachable!()` panics and their format bloat with this single one. As you say, the only real alternative is a fallible iterator that would be used with `let mut it = m.iter_paths(); while let Some(ele) = it.next()? { ... }`.
But it only moves having to deal with the error up one level. Because of that, I think re-evaluating the mqtt client with its numerous unwraps() is a lower hanging fruit.

Reply:
I'm not sure that `panic!()` and `unreachable!()` have the same panic bloat as an `unwrap()`. If the user specifies `panic` as an `abort`, you don't get format bloat, but rather just panic data bloat (metadata about where the panic occurred) in RODATA.

Reply:
Given the more annoying DX of a fallible iterator, I'd lean towards (in a future PR when we re-evaluate the MQTT client) add that in parallel to the existing panicing one.

Reply:
> If the user specifies panic as an abort, you don't get format bloat (I think?),

Right. Same in both cases.

https://doc.rust-lang.org/src/core/result.rs.html#1817-1827